CreeDictionary/phrase_translate/management/commands/translatewordforms.py

```diff
@@ -0,0 +1,141 @@
+        self._buffer = []
+
+
+class Command(BaseCommand):
+    help = """Translate wordforms into English phrases
+    
+    This command:
+      - iterates over all wordforms in the database
+      - uses the wordform inflection analysis and the phrase-translator FSTs to
+        auto-generate phrase definitions from the definitions of the
+        corresponding lemmas
+      - saves these auto-generated wordform definitions to the database with
+        source name “auto”
+    """
+
+    def add_arguments(self, parser: ArgumentParser):
+        parser.add_argument("--log-level")
+
+    def handle(self, *args, **options):
+        if options["log_level"]:
+            logger.setLevel(options["log_level"])
+
+        (ds, created_) = DictionarySource.objects.get_or_create(abbrv="auto")
+        Definition.objects.filter(citations__abbrv="auto").delete()
+
+        definitions = defaultdict(set)
+        for d in Definition.objects.all().iterator():
+            definitions[d.wordform_id].add(d)
+        defn_count = sum(
+            len(definition_list) for definition_list in definitions.values()
+        )
+        logger.info(f"loaded {defn_count} definitions")
+
+        definition_buffer = InsertBuffer(manager=Definition.objects)
+        citation_buffer = InsertBuffer(manager=Definition.citations.through.objects)
+
+        # Some status counts to report at the end:
+        # no translation returned, typically because no +N or +V tag
+        no_translation = 0
+        # foma returned n != 1 analyses
+        error_count = 0
+
+        wordform_count = Wordform.objects.count()
+        for w in tqdm(
+            Wordform.objects.select_related("lemma").iterator(), total=wordform_count
+        ):
+            for definition in definitions.get(w.lemma_id, []):
+                try:
+                    if not w.analysis:
+                        continue
+
+                    tags = parse_analysis_and_tags(w.analysis)
+                    phrase = inflect_english_phrase(tags, definition.text)
+
+                    if not phrase:
+                        logger.debug(f"no translation for {w.text} {tags}")
+                        no_translation += 1
+                        continue
+
+                    auto_definition = Definition(
+                        text=phrase,
+                        wordform_id=w.id,
+                        auto_translation_source_id=definition.id,
+                    )
+
+                    definition_buffer.add(auto_definition)
+                    citation_buffer.add(
+                        Definition.citations.through(
+                            dictionarysource_id=ds.abbrv,
+                            definition_id=auto_definition.id,
+                        )
+                    )
+                except KeyError as e:
+                    logger.debug(
+                        f"Unknown tag {e.args[0]} for {w.text} {w.analysis}", e
+                    )
+                    raise
```

Comment:
My `try/except` things were so expansive because I did not realize that, unlike Java, `try: … except: ` in Python *doesn’t* introduce a new scope. Fixed now.